Retire amendment fix1515

pratikmankawde opened the pull request:
## High Level Overview of Change
This change retires the `fix1515` amendment.

### Context of Change
Amendments activated for more than 2 years can be retired.

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Performance (increase or change in throughput and/or latency)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [ ] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

a1q123456 commented on src/xrpld/app/paths/detail/BookStep.cpp:
```diff
@@ -85,9 +85,7 @@ class BookStep : public StepImp<TIn, TOut, BookStep<TIn, TOut, TDerived>>
     static uint32_t
     getMaxOffersToConsume(StrandContext const& ctx)
```
ctx is not referenced so we can remove the name making it `StrandContext const&` only

a1q123456 replied:
Actually, it's only used once in BookStep.cpp, I think we can just define a constexpr variable in the cpp file to replace this function

a1q123456 replied:
Actually, we should just make `maxOffersToConsume_` a static constexpr variable

bthomee replied:
I agree with the above, having a function just for returning a constant value can be simplified.

pratikmankawde replied:
Done

a1q123456 reviewed (approved):
LGTM

a1q123456 commented:
looks good! feel free to add a ready to merge label

bthomee committed:
refactor: Retire fix1515 amendment

Amendments activated for more than 2 years can be retired. This change retires the fix1515 amendment.
